packages/utilities/src/ImperativeConfig.ts

```diff
@@ -301,4 +302,21 @@ export class ImperativeConfig {
             ImperativeConfig.instance.loadedConfig.profiles.forEach(profile => schemas[profile.type] = profile.schema);
         return schemas;
     }
+
+    /**
+     * Checks if obj complies with the { @link ICredentialManager } interface
+     * @throws ImperativeError if object is of type ICredentialManager but does not have required 'type' property
+     * @returns boolean representing whether it complies or not with the interface
+     */
+    public isCredentialManager(obj: any): obj is ICredentialManager {
```

Comment:
It needs to have the `obj is ICredentialManager` in order to check if the object also complies with the interface in order to prevent any kind of object from being passed in and assure an object that has the plugin and type properties.

Reply:
Read more here: https://stackoverflow.com/questions/14425568/interface-type-check-with-typescript